stability-tests/many-tips/main.go

```diff
@@ -271,13 +255,23 @@ func mineLoopUntilHavingOnlyOneTipInDAG(rpcAddress string, miningAddress util.Ad
 	if err != nil {
 		return errors.Wrapf(err, "Error in getCurrentTipsLength")
 	}
-	for numOfTips > 1 {
+	hasTimedOut := false
+	for numOfTips > 1 && !hasTimedOut {
 		time.Sleep(1 * time.Second)
 		numOfTips, err = getCurrentTipsLength(rpcClient)
 		if err != nil {
 			return errors.Wrapf(err, "Error in getCurrentTipsLength")
 		}
 	}
+	spawn("ChecksIfTimeIsUp", func() {
+		timer := time.NewTimer(20 * time.Minute)
+		<-timer.C
+		hasTimedOut = true
+	})
+
+	if hasTimedOut {
```

Comment:
hasTimedOut will never actually be true, since the routine is not spawned until _after_ the for loop.